### ISSUE TITLE
Ideas for the GUI rewrite!

### ISSUE THREAD

**Issue by brittyazel** [brittyazel/Neuron#49]
Please share all your ideas for the GUI rewrite that is in progress. This is the place to point out any places where you feel Neuron has shortcomings, and could be improved

**Comment by teh-fixerer** [comment]
- For healer bars it might be a useful feature to be able to paste in a character name from your raid or party list, and then all your heal spells can use that as their target.  (Neuron/Ions predecessor Macaroon had that feature)

- Would be nice to have the pet bar, status bars and bag bars part of the character profile.  I do this manually by editing the .toc files at update time and changing the lines for example.
## SavedVariables: 
## SavedVariablesPerCharacter: NeuronPetCDB, NeuronPetGDB

**Comment by pottervogel** [comment]
well, you asked ^^ and I see you are working on the flyouts, so maybe you could even improve those by giving us the possibility to use TSM (TradeSkillMaster) groups or ArkInventory filters for the flyout buttons. both are well maintained and pretty common. but TSM4 is in alpha and i don't know if they are changing their formats

**Comment by pottervogel** [comment]
another thing that would be awysome is to import single bars to profiles, but i think that may be tricky

**Comment by pottervogel** [comment]
the macro edit frame could be a little bigger or resizable

**Comment by brittyazel** [comment]
You can get ride of the graphics on the zone and extra bars in the bar editor. It's a toggle

**Comment by brittyazel** [comment]
> I have a lot of trouble manually aligning bars when snap-to interferes... I thought there used to be (maybe Macaroon days, maybe more recently) a way to very incrementally way to move a bar via shift-arrow or something but I haven't been able to get that working lately and manually adjusting the positioning numbers is annoying.

The best we can do for this at the moment is the X and Y position arrows. I changed them so they move 1px at a time now, rather than .1 pixel at a time (whoever set that to be the default is a psychopath)

**Comment by brittyazel** [comment]
> I'd like the option for the button tooltips to use the main tooltip location (I use an add-on called TipTac to move it from the default location) rather than happening at the button itself.

Yeah that was a recent change actually, I should add that to a toggle or something so you can set where you want it to go

**Comment by brittyazel** [comment]
> I'd like the option to actually show the button icon for something that's not currently available rather than just blanking it (I get it for situations where the spell isn't actually in the spellbook, like with a talent change, since you may not be able to GET the icon, although that'd be nice too... but at minimum I'm looking for situations where it is in the spell book)... thinking something like Mend Pet when you don't actually have a pet active (flying, etc), I don't see the benefit of it blanking out in that scenario.

solid point. I wasn't aware that was the case. I knew that it would go blank when switching specs and losing the ability (because we can no longer query the icon), but I didn't know that "unavailable" buttons also went blank

**Comment by brittyazel** [comment]
> Would it be possible for some basic mouseover documentation about the various config options? I've been using this add-on since the Macaroon days (I just missed Trinity) and I still have no idea what things like Spell Alerts or AlphaUp are or the Reaction bar state... I have theories and worst case I could just play around with them but I'd love to see something even quick and dirty when hovering over the options.

Good idea! I can do something like that with the new GUI, the old GUI is kinda stuck as-is for the moment. It seems like changing one thing can bring the whole house of cards down.

**Comment by brittyazel** [comment]
> I'm assuming this isn't possible but occasionally when editing bars I get into combat (back aggro, trying to put a potion on the bar with 2 seconds left in a pull timer, etc) and I can't switch out of edit mode so the buttons are useable and I'm stuck working out of my spellbook... being able to get out of edit mode (or better, have it do it automatically when in combat) would be incredibly helpful.
agreed

**Comment by brittyazel** [comment]
> I'm pretty sure it used to be possible (again, back in Macaroon-era days) to drag bag items onto action bars even when they weren't useful (for instance, you're collecting 100 skins for a crafted item and you want to see how many you've collected), that went away at some point but I found it occasionally very useful to do. It just showed the quantity as the stack number but didn't do anything if you clicked on it.

I think that was back before the buttons were all treated as macros. Back in the day I think you could choose if a button was an action button or a macro button. The problem with dragging an item that doesn't have a use on a bar is that there's no spellID for it. Thus it can't be made into a macro. Its actually a much more complicated issue than it appears on the surface

**Comment by raysmith59** [comment]
Yeah, these are meant for the new GUI specifically.

The "show button icon" one was already reported by someone on #101, just noticed it now.

You're right, the buttons used to self-select between action and macro buttons, I'd completely forgotten about that.  Well, consider it on my wish list without much expectation that it'll happen, I can probably figure out a way to track via TMW although that'd require more setup than just dragging an icon on to a button.

If the graphics toggle is in the editor then how can I do that?  When I put Neuron in edit mode those buttons don't appear.  Do the buttons need to be active to do that?

**Comment by brittyazel** [comment]
Oh ok, maybe I'm mistaken then. I'll see what I can do

**Comment by raysmith59** [comment]
* I'd also like the ability to set the colour options per-button as opposed to per-bar, there are some buttons where I care about showing duration and cooldown, others where I only care about the cooldown.
